utilidades/storage_file_utils.py

```
@@ -500,92 +541,49 @@ def _add_partition_cols(self, df: pd.DataFrame, mercado: str) -> pd.DataFrame:
+                    
+                    # Combine and write
+                    self._combine_and_write_partition(
+                        new_data_df=new_data_partition_df,
+                        existing_data_df=existing_data_df,
+                        partition_cols=partition_cols,
+                        output_file_path=output_file_path,
+                        value_cols=value_cols,
+                        dataset_type=dataset_type
+                    )
+                except Exception as e:
+                    raise Exception(f"Failed to process partition {partition.to_dict()}: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)